### ISSUE TITLE
How to build everything using our own version of GCC ?

### ISSUE THREAD

**Issue by jiapei100** [gnu-mcu-eclipse/arm-none-eabi-gcc-build#5]
Just wonder if we can build the source from scratch using our own version of GCC? And the release version number **8.2.1-1.7** seems to be corresponding to GCC version? Does that mean, so far, it's **ONLY** compatible with **GCC 8.2.1-1.7**? However, my GCC is **ALREADY** **GCC 8.3.0**, can I still build this source?

**Comment by ilg-ul** [comment]
> build the source from scratch using our own version of GCC

yes, the script is public, you can fork the project and tweak it to do whatever you want

> 8.2.1-1.7 seems to be corresponding to GCC version?

that's correct, it is based on GCC 8.2.1, the commits used by the first ARM release, and it is the 7th release from GNU MCU Eclipse (it took a while to fix lots of bugs).

> my GCC is ALREADY GCC 8.3.0, can I still build this source?

I strictly follow ARM releases, and I have no experience with more recent versions. you probably can, but how can you tell for sure the compiler is functioning properly? ARM is also running many validation tests, so when I use their sources I expect my build to be also functional.
